From 0e47b62aa84657fb23bc342ab41d5e28021881ac Mon Sep 17 00:00:00 2001
From: Ilies CHERGUI <ichergui@nvidia.com>
Date: Tue, 8 Jul 2025 19:04:21 +0100
Subject: [PATCH 3/7] v4l2bufferpool: Avoid set_flushing warning

The gst_buffer_pool_set_flushing() warns when that function is called
on an inactive pool. Avoid the warning by checking the state, this is
similar to what we do in gst_v4l2_object_unlock().

Part-of: <https://gitlab.freedesktop.org/gstreamer/gst-plugins-good/-/merge_requests/649>

Upstream-Status: Backport [https://gitlab.freedesktop.org/gstreamer/gst-plugins-good/-/merge_requests/649]
Signed-off-by: Jose Quaresma <quaresma.jose@gmail.com>
Signed-off-by: Ilies CHERGUI <ichergui@nvidia.com>
---
 gstv4l2bufferpool.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/gstv4l2bufferpool.c b/gstv4l2bufferpool.c
index bf02884..0bc2caa 100644
--- a/gstv4l2bufferpool.c
+++ b/gstv4l2bufferpool.c
@@ -1345,7 +1345,7 @@ gst_v4l2_buffer_pool_flush_start (GstBufferPool * bpool)
   g_cond_broadcast (&pool->empty_cond);
   GST_OBJECT_UNLOCK (pool);
 
-  if (pool->other_pool)
+  if (pool->other_pool && gst_buffer_pool_is_active (pool->other_pool))
     gst_buffer_pool_set_flushing (pool->other_pool, TRUE);
 
   GST_DEBUG_OBJECT (pool, "End flushing");
@@ -1358,7 +1358,7 @@ gst_v4l2_buffer_pool_flush_stop (GstBufferPool * bpool)
 
   GST_DEBUG_OBJECT (pool, "stop flushing");
 
-  if (pool->other_pool)
+  if (pool->other_pool && gst_buffer_pool_is_active (pool->other_pool))
     gst_buffer_pool_set_flushing (pool->other_pool, FALSE);
 
 #ifndef USE_V4L2_TARGET_NV
-- 
2.34.1

